test/integration/test/recipe_test.js

```diff
@@ -0,0 +1,177 @@
+            'range':{'Eq':{'OpenClosed':'Open'}}
+          }],
+          'execute':
+            [{'destination':
+              [{'kind':'Log'}],
+              'kind':'Log',
+              'value':{'String':'Closed'}}]}
+            ]};
+source = JSON.stringify(source);
+
+function generateThinkerbellNewRecipePayload(recipeName) {
+  return {
+    'select':{
+       'kind':'AddThinkerbellRule'
+    },
+    'value':{
+       'ThinkerbellRule':{
+          'name': recipeName,
+          'source': source
+        }
+    }
+  };
+}
+
+Prepper.makeSuite('Add/Remove recipe',function(){
+
+  before(Prepper.foxboxManager.foxboxLogin);
+
+  it('Check service list for recipe',function(){        
+    return chakram.get(Prepper.foxboxManager.serviceListURL)
+    .then(function(listResponse) {
+      expect(listResponse).to.have.status(200);
+      var isFound = listResponse.body.some(
+        entry => entry.adapter === 'thinkerbell@link.mozilla.org');
+      // .equals(true) passes if ['1'] is returned for instance, IIRC
+      expect(isFound).to.be.true;
+    });
+  });
+
+  describe('add recipes', function(){
+    it('Add recipe',function(){
+    
+    return chakram.put(Prepper.foxboxManager.setterURL, 
+      generateThinkerbellNewRecipePayload('First Recipe'))
+      .then(function(cmdResponse) {
+        expect(cmdResponse).to.have.status(200);
+        expect(cmdResponse.body['thinkerbell-add-rule']).equals(null);
+        return chakram.get(Prepper.foxboxManager.serviceListURL);
+      })
+      .then(function(listResponse) {
+        expect(listResponse).to.have.status(200);
+        var isFound = listResponse.body.some(
+          entry => entry.id === 'thinkerbell/First Recipe');
+        expect(isFound).to.be.true;
+      });
+    }); 
+
+    it('Add two recipes one by one',function(){        
+      
+      return chakram.put(Prepper.foxboxManager.setterURL, 
+        generateThinkerbellNewRecipePayload('Second Recipe'))
+        .then(function(cmdResponse) {
+
+          expect(cmdResponse).to.have.status(200);
+          expect(cmdResponse.body['thinkerbell-add-rule']).equals(null);
+          return chakram.put(Prepper.foxboxManager.setterURL, 
+            generateThinkerbellNewRecipePayload('Third Recipe'));
+        })
+        .then(function(cmdResponse) {
+          expect(cmdResponse).to.have.status(200);
+          expect(cmdResponse.body['thinkerbell-add-rule']).equals(null);
+          return chakram.get(Prepper.foxboxManager.serviceListURL);
+        })
+        .then(function(listResponse) {
+          expect(listResponse).to.have.status(200);
+          var isFound = listResponse.body.some(
+            entry => entry.id === 'thinkerbell/Second Recipe');
+          expect(isFound).to.be.true; 
+          isFound = listResponse.body.some(
+            entry => entry.id === 'thinkerbell/Third Recipe');
+          expect(isFound).to.be.true; 
+        });
+      });
+  
+    after(function(){
+      var payload = {'select':
```

Comment:
Nit: This can be reduced to something smaller, like:
```js
var promises = ['First', 'Second', 'Third'].map(number => chakram.put(Prepper.foxboxManager.setterURL, {'select':
  {'id': 'thinkerbell/' + number + ' Recipe/remove'},'value':null});
return Promise.all(promises);
```